src/program.ts

```diff
@@ -3398,7 +3474,29 @@ export class Function extends TypedElement {
   lookup(name: string): Element | null {
     var locals = this.localsByName;
     if (locals.has(name)) return assert(locals.get(name));
-    return this.parent.lookup(name);
+    var parentResult = this.parent.lookup(name);
+    if(parentResult && this.parent.kind == ElementKind.FUNCTION) {
+      var parentFunction = <Function>this.parent;
+      if(parentFunction.closedLocals.size > 0) { //TODO allow nested closure definitions
+        this.program.error(
+          DiagnosticCode.Not_implemented,
+          this.identifierNode.range, this.identifierNode.text
+        );
+        return null;
+      }
+      if(parentResult.kind == ElementKind.LOCAL) {
+        let local = changetype<Local>(parentResult)
+        if(this.closedLocals.has(local.name)) return assert(this.closedLocals.get(local.name))
+        let mask = local.type.byteSize - 1;
+        let memoryOffset = this.nextGlobalClosureOffset;
+        if (memoryOffset & mask) memoryOffset = (memoryOffset | mask) + 1;
+        var closedLocal = local.close(memoryOffset);
+        this.nextGlobalClosureOffset += this.nextGlobalClosureOffset;
```

Comment:
Replicating the logic used to determine field offsets at proper byte alignment

Reply:
This is looking fine